storage/interface.go

```diff
@@ -282,7 +282,18 @@ type HistogramAppender interface {
 	// For efficiency reasons, the histogram is passed as a
 	// pointer. AppendHistogram won't mutate the histogram, but in turn
 	// depends on the caller to not mutate it either.
-	AppendHistogram(ref SeriesRef, l labels.Labels, t int64, h *histogram.Histogram, fh *histogram.FloatHistogram) (SeriesRef, error)
+	//
+	// By default, the storage should assume the histogram to be a counter histogram
+	// unless the hints say it is a gauge histogram.
+	AppendHistogram(ref SeriesRef, l labels.Labels, t int64, h *histogram.Histogram, fh *histogram.FloatHistogram, hints *AppendHints) (SeriesRef, error)
+}
+
+// AppendHints are the hints given to the storage
+// about the sample being appended.
+type AppendHints struct {
```

Comment:
Why we pass struct instead just a bool? Do we plan to pass more hints in the future?

Reply:
I guess so. We could add hints about counter resets, for example. But @codesome might have had more thoughts.

Reply:
Yes, more hints in the future possibly. See https://github.com/prometheus/prometheus/issues/11765. It is easier this way where you only have to refactor once.